orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/EvaluateDeploymentHealthTask.java

```diff
@@ -0,0 +1,105 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.monitoreddeploy;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.config.DeploymentMonitorServiceProvider;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.EvaluateHealthRequest;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.EvaluateHealthResponse;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.StatusReason;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(value = "monitored-deploy.enabled")
+public class EvaluateDeploymentHealthTask extends MonitoredDeployBaseTask {
+  @Autowired
+  EvaluateDeploymentHealthTask(
+      DeploymentMonitorServiceProvider deploymentMonitorServiceProvider, Registry registry) {
+    super(deploymentMonitorServiceProvider, registry);
+  }
+
+  @Override
+  public TaskResult executeInternal() {
+    EvaluateHealthRequest request = new EvaluateHealthRequest(stage);
+
+    log.info(
+        "Evaluating health of deployment {} at {}% with {}",
+        request.getNewServerGroup(), request.getCurrentProgress(), monitorDefinition);
+
+    EvaluateHealthResponse response = monitorDefinition.getService().evaluateHealth(request);
+    sanitizeAndLogResponse(response);
+
+    EvaluateHealthResponse.NextStepDirective directive = response.getNextStep().getDirective();
+
+    Id statusCounterId =
+        registry
+            .createId("deploymentMonitor.healthStatus")
+            .withTag("monitorId", monitorDefinition.getId())
+            .withTag("status", directive.toString());
+
+    registry.counter(statusCounterId).increment();
+
+    if (directive != EvaluateHealthResponse.NextStepDirective.WAIT) {
+      // Log timing
+      Id timerId =
+          registry
+              .createId("deploymentMonitor.timing")
+              .withTag("monitorId", monitorDefinition.getId())
+              .withTag("status", directive.toString());
+
+      double duration = Instant.now().toEpochMilli() - stage.getStartTime();
+      registry.gauge(timerId).set(duration);
+    }
+
+    List<StatusReason> statusReasons =
+        Optional.ofNullable(response.getStatusReasons()).orElse(Collections.emptyList());
+
+    TaskResult.TaskResultBuilder taskResultBuilder;
+
+    switch (directive) {
```

Comment:
I think the mapping from directive to task status is neat but could be expressed more clearly factored out. Something like this: 

```
  TaskResultBuilder processDirective(EvaluateHealthResponse.NextStepDirective directive) {
    switch(directive) {
      // TODO(mvulfson): Actually implement this case in the stages
      case COMPLETE: return TaskResult.builder(ExecutionStatus.SUCCEEDED).output("skipToPercentage", 100);
      case CONTINUE: return TaskResult.builder(ExecutionStatus.SUCCEEDED);
      case WAIT: return TaskResult.builder(ExecutionStatus.RUNNING);
      case ABORT: return TaskResult.builder(ExecutionStatus.TERMINAL);
      default: return TaskResult.builder(ExecutionStatus.TERMINAL);
  }
}    
```